fastcrawler/engine/aio.py

```diff
@@ -128,69 +127,78 @@ async def teardown(self) -> None:
 
     async def base(
         self,
-        url: pydantic.AnyUrl,
-        method: str,
-        data: dict | None,
-        headers: dict | None = None,
-        cookies: dict | None = None,
+        request: Request,
         verify_ssl=False,
-    ) -> Response | None:
+    ) -> RequestCycle | None:
         """Base Method for protocol to retrieve a list of URL."""
+        assert request.method is not None
+
         if self.session:
+            if isinstance(request.data, dict):
+                json = request.data
+                data = None
+            else:
+                json = None
+                data = request.data
+
             async with self.session.request(
-                method,
-                str(url),
+                request.method,
+                request.url,
+                json=json,
                 data=data,
-                headers=headers or self.headers,
-                cookies=self._get_morsel_cookie(cookies) if cookies else self.cookies,
-                verify_ssl=verify_ssl,
+                headers=request.headers or self.headers,
+                cookies=self._get_morsel_cookie(request.cookies)
+                if request.cookies
+                else self.cookies,
+                ssl=verify_ssl,
                 **self._proxy,
             ) as response:
-                return await self.translate_to_response(response, url)
+                return await self.translate_to_response(response, request)
         return None
 
-    async def batch(self, requests: list[Request], method: str) -> dict[str, Response]:
+    async def batch(self, requests: list[Request], method: str) -> dict[Url, RequestCycle]:
         """Batch Method for protocol to retrieve a list of URL."""
         for request in requests:
             request.method = method
-        tasks = {request.url: None for request in requests}
-        results = await asyncio.gather(
-            *[
-                self.base(
-                    request.url,
-                    request.method,
-                    data=request.data,
-                    headers=request.headers,
-                    cookies=request.cookies,
-                )
-                for request in requests
-            ]
-        )
-        return {url: result for url, result in zip(tasks.keys(), results)}
+        tasks = []
```

Comment:
use list() instead of []

Reply:
that's not correct. using  language construct is always better. 
Linters prefer  language construct 
and also it's better in term of perfomance